testsuite/do_tests.sh

```diff
@@ -456,23 +461,23 @@ fi
 echo
 echo "Phase 7: Running PyNEST tests"
 echo "-----------------------------"
+
 if test "${PYTHON}"; then
     PYNEST_TEST_DIR="${TEST_BASEDIR}/pytests/"
     XUNIT_NAME="07_pynesttests"
+    
+    # Run all tests except those in the mpi subdirectory
     XUNIT_FILE="${REPORTDIR}/${XUNIT_NAME}.xml"
-    "${PYTHON}" "${NOSE}" -v --with-xunit --xunit-testsuite-name="${XUNIT_NAME}" \
-		--xunit-file="${XUNIT_FILE}" --exclude=test_mpitests\.py "${PYNEST_TEST_DIR}" 2>&1 \
-        | tee -a "${TEST_LOGFILE}" | grep --line-buffered "\.\.\. ok\|fail\|skip\|error" | sed 's/^/  /'
-
-    if test "${HAVE_MPI}" = "true"; then
-	echo
-	echo "  Running PyNEST tests with MPI (no output will be produced)"
-	XUNIT_NAME="${XUNIT_NAME}_mpi"
-	XUNIT_FILE="${REPORTDIR}/${XUNIT_NAME}.xml"
-	"${PYTHON}" "${NOSE}" -v --with-xunit --xunit-testsuite-name="${XUNIT_NAME}" \
-		    --xunit-file="${XUNIT_FILE}" "${PYNEST_TEST_DIR}/test_mpitests.py" \
-		    2>&1 | tee -a "${TEST_LOGFILE}" >/dev/null
-	            # "&>FILE" or ">>FILE 2>&1" don't silence the line above. Why?!
+    "${PYTEST}" --verbose --junit-xml="${XUNIT_FILE}" --numprocesses=auto \
+          --ignore="${PYNEST_TEST_DIR}/mpi" "${PYNEST_TEST_DIR}" 2>&1 | tee -a "${TEST_LOGFILE}" 
+  
+    # Run tests in the mpi subdirectories, grouped by number of processes
+    if test "${HAVE_MPI}" = "true" -a "${MPIRUN}" ; then
+       for numproc in 2 4 ; do
```

Comment:
Maybe you can make this iterate over all existing directories? That way, future extensions would really just require to drop a file or symlink into a directory.
```suggestion
       for numproc in $(cd ${PYNEST_TEST_DIR}/mpi/; ls -d */ | tr -d '/'); do
```